plugins/generic/metadataExport/MetadataExportHandler.inc.php

```diff
@@ -0,0 +1,167 @@
+	 * @param $args Array
+	 */
+	function export($args) {
+		$scope = Request::getUserVar('exportScope');
+		
+		$this->exportPluginName = Request::getUserVar('exportPluginName');
+		$this->elementId = Request::getUserVar('elementId');
+		$this->metadataExportPlugins = PluginRegistry::loadCategory('generic/metadataExport/metadataExportFormats');
+		$this->exportPlugin = $this->metadataExportPlugins[$this->exportPluginName];
+		$this->journal = Request::getJournal();
+		$this->publishedArticleDao = DAORegistry::getDAO('PublishedArticleDAO');
+		$this->issueDao = DAORegistry::getDAO('IssueDAO');
+
+		switch ($scope) {
+			case 'exportArticle':
+				$this->_exportArticle();
+				break;
+			case 'exportIssue':
+				$this->_exportIssue();
+				break;
+			case 'exportJournal':
+				$this->_exportJournal();
+				break;
+			default:
+				assert(false);
+		}
+	}
+	
+	/**
+	 * @function _exportArticle - Export current article
+	 */
+	function _exportArticle() {
+		$articleId = $this->elementId;
+		$article = $this->publishedArticleDao->getPublishedArticleByArticleId($articleId);
+		$issue = $this->issueDao->getIssueById($article->getIssueId());
+		$this->_addIssueData($article, $issue);
+		
+		$this->fileName = $article->getLocalizedTitle();
+		$this->articles = array($article);
+		$this->_createFile();
+	}
+	
+	/**
+	 * @function _exportIssue - Export current issue
+	 */
+	function _exportIssue() {
+		$issueId = $this->elementId;
+		$issue = $this->issueDao->getIssueById($issueId);
+		$this->articles = $this->publishedArticleDao->getPublishedArticles($issueId);
+		
+		foreach($this->articles as $article) {
+			$this->_addIssueData($article, $issue);
+		}
+		
+		$this->fileName = $issue->getLocalizedTitle();
+		$this->_createFile();
+	}
+	
+	/**
+	 * @function _exportJournal - Export current journal
+	 */
+	function _exportJournal() {
+		$journalId = $this->journal->getId();
+		
+		$articleIterator = $this->publishedArticleDao->getPublishedArticlesByJournalId($journalId);
+		$this->articles = array();
+		
+		while ($article = $articleIterator->next()) {
+			$issueId = $article->getIssueId();
+			$issue = $this->issueDao->getIssueById($issueId);
+			$this->_addIssueData($article, $issue);
+			
+			$this->articles[] = $article;
+		}
+		unset($articleIterator);
+		
+		$this->fileName = $this->journal->getLocalizedTitle();
+		$this->_createFile();
+	}
+	
+	/**
+	 * @function _createFile - Create text or xml file
+	 */
+	function _createFile() {
+		if ($this->exportPlugin->isXML()) {
+			$this->exportPlugin->createXmlFile($this->journal, $this->articles, $this->fileName);
+		} else {
+			$this->exportPlugin->createTextFile($this->journal, $this->articles, $this->fileName);
+		}
+	}
+	
+	/**
+	 * @function _addIssueData - Add some issue data to the article object
+	 * @param $article ArticleDAO object
+	 * @param $issue IssueDAO object
+	 */
+	function _addIssueData($article, $issue) {
```

Comment:
Is this being done because the Issue object isn't available somewhere? IMO it would be nicer to let the issue describe itself, rather than injecting issue metadata into the article object.